code/modules/client/client procs.dm

```diff
@@ -438,6 +434,34 @@
 
 	initialized = TRUE
 
+/client/proc/InitHolder()
+	
+	var/assign_holder = admin_datums[ckey]
+
+	if(config.admin_restrict_by_computerid && !config.admin_legacy_system && assign_holder) //Restrict by computerid only works with the SQL System enabled
+		if(!establish_db_connection(dbcon))
+			log_debug("Holder Assignment: SQL ERROR - Failed to connect. - Falling back to non-whitelist")
+		else
+			//Query the database for the computerid associated with that admin
+			var/DBQuery/cid_ckey_query = dbcon.NewQuery("SELECT count(*) FROM ss13_admin_computerid WHERE ckey = :ckey: AND computerid = :computerid:")
+			cid_ckey_query.Execute(list("ckey"=ckey,"computerid"=computer_id))
+			var/count = 0
+			if (cid_ckey_query.NextRow())
+				count = text2num(cid_ckey_query.item[1])
+			//If we cant find them in the db, dont assign them any permissions
+			if(!count)
+				to_chat(src,"<span class='danger'>This computer is not associated with your admin account. No permissions have been assigned!</span>")
+				assign_holder = null
+				spawn(0)
```

Comment:
I wonder if a spawn/delay here is necessary. Tho I guess since the discord API can be slow at times/time out completely.

Also, addtimer over spawn. You can use two timers.